app/screens/RootNavigator.tsx

```diff
@@ -20,6 +21,7 @@ export function RootNavigator (): JSX.Element {
   return (
     <WalletProvider data={wallets[0]}>
       <WalletAddressProvider>
+        <TransactionAuthorization />
```

Comment:
This one shouldn't be imported if the wallet is not encrypted. The problem is it's causing the UI to break if the current loaded wallet is an unencypted one. 

For iOS, even if you uninstall the app, the localStorage data is still there. Previous users who tested the unencrypted wallet will not be able to test since they will have infinite loop of encypted error.

Reply:
This component should be nested within `MnemonicEncryptedProvider`

https://github.com/DeFiCh/wallet/pull/403/files#diff-6401c3d7908d3b27690269213369a1ad75c5b40f42260e28db3a515b14dc6e59R100

Reply:
@thedoublejay technically it is a dumb component.

From EncryptedWalletProvider perspective, this is a prompting UI to collect a 6 digits string.

From TransactionAuth component perspective
```
// const encryptionUI = useEncryptedWalletUI() // this is not really this component concern
const { pinConsumer } = props
pinConsumer.provide(this) // myself

onPrompt () => {
  // wait for user
  return userInput // doesnt matter what prompt consumer do this with
}
```

Reply:
> For iOS, even if you uninstall the app, the localStorage data is still there. Previous users who tested the unencrypted wallet will not be able to test since they will have infinite loop of encypted error.

Wallet should be wiped after 4 attempts and see onboarding flow again.
Hmm this is more like UX to solve issue. If an iphone ownership somehow transferred from one light wallet user to another. (later there should be implementation: when app focus opened/focused, prompt for passphrase) Maybe there is an option "Not my wallet, wipe and re-onboard"

Reply:
@ivan-zynesis it's not even related to components. There's an error thrown on the useEffect. It's due to having an unencrypted wallet loaded